src/Tests/Microsoft.NET.ToolPack.Tests/GivenThatWeWantToPackAToolProject.cs

```diff
@@ -0,0 +1,87 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Xml.Linq;
+using Microsoft.NET.TestFramework;
+using Microsoft.NET.TestFramework.Assertions;
+using Microsoft.NET.TestFramework.Commands;
+using FluentAssertions;
+using Xunit;
+using Xunit.Abstractions;
+using NuGet.Packaging;
+
+namespace Microsoft.NET.ToolPack.Tests
+{
+    public class GivenThatWeWantToPackAToolProject : SdkTest
+    {
+        private readonly string _nugetPackage;
+
+        public GivenThatWeWantToPackAToolProject(ITestOutputHelper log) : base(log)
+        {
+            TestAsset helloWorldAsset = _testAssetsManager
+                .CopyTestAsset("PortableTool", "PackPortableTool")
+                .WithSource()
+                .Restore(Log);
+
+            var packCommand = new PackCommand(Log, helloWorldAsset.TestRoot);
+
+            packCommand
+                .Execute()
+                .Should()
+                .Pass();
+
+            _nugetPackage = packCommand.GetNuGetPackage();
+        }
+
+        [Fact]
+        public void It_packs_successfully()
+        {
+            using (var nupkgReader = new PackageArchiveReader(_nugetPackage))
```

Comment:
I know we used to assert on intermediate nuspec. However, in my case, a lot of the production logic can be done by nuspec as well. So I feels this is circular and not reliable. So I choose to use "real" nupkg and a package from NuGet to assert the content

Reply:
I don't have a problem with this. Maybe we should fix the existing pack test to use the same pattern.